static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,76 @@
+import {createContext, useCallback, useContext, useEffect, useState} from 'react';
+
+import {DrawerBody, DrawerPanel} from 'sentry/components/globalDrawer/components';
+import type {
+  DrawerConfig,
+  DrawerContext as TDrawerContext,
+} from 'sentry/components/globalDrawer/types';
+import {useLocation} from 'sentry/utils/useLocation';
+
+const DEFAULT_DRAWER_CONFIG: DrawerConfig = {
+  renderer: null,
+  options: {
+    closeOnEscapeKeypress: true,
+    closeOnOutsideClick: true,
+  },
+};
+
+const DEFAULT_DRAWER_CONTEXT: TDrawerContext = {
+  openDrawer: () => {},
+  closeDrawer: () => {},
+};
+
+const DrawerContext = createContext<TDrawerContext>(DEFAULT_DRAWER_CONTEXT);
+
+export function GlobalDrawer({children}) {
+  const location = useLocation();
+  const [drawerConfig, setDrawerConfig] = useState<DrawerConfig>(DEFAULT_DRAWER_CONFIG);
+  const openDrawer = useCallback(
+    (renderer, options = {}) => setDrawerConfig({renderer, options}),
+    []
+  );
+  const closeDrawer = useCallback(() => setDrawerConfig(DEFAULT_DRAWER_CONFIG), []);
+  const drawerContextValue: TDrawerContext = {
+    closeDrawer,
+    openDrawer,
+  };
+  const {renderer, options = {}} = drawerConfig;
+  const {closeOnEscapeKeypress = true, closeOnOutsideClick = true} = options;
+
+  const handleClose = useCallback(() => {
+    // Callsite callback when closing the drawer
+    options?.onClose?.();
+    // Actually close the drawer component
+    closeDrawer();
+  }, [options, closeDrawer]);
+
+  // Close the drawer when the browser history changes.
+  useEffect(() => closeDrawer(), [location.pathname, closeDrawer]);
+  const isDrawerOpen = typeof renderer === 'function';
+  const renderedChild =
+    renderer?.({
+      Body: DrawerBody,
+      closeDrawer: handleClose,
+    }) ?? null;
+
+  return (
+    <DrawerContext.Provider value={drawerContextValue}>
+      {renderer !== null && (
+        <DrawerPanel
+          isOpen={isDrawerOpen}
+          onClose={handleClose}
+          onOpen={options?.onOpen}
+          closeOnOutsideClick={closeOnOutsideClick}
+          closeOnEscapeKeypress={closeOnEscapeKeypress}
+        >
+          {renderedChild}
+        </DrawerPanel>
+      )}
+      {children}
+    </DrawerContext.Provider>
+  );
```

Comment:
```suggestion
  const isDrawerOpen = renderer !== null;
  const renderedChild =
    renderer?.({
      Body: DrawerBody,
      closeDrawer: handleClose,
    }) ?? null;

  return (
    <DrawerContext.Provider value={drawerContextValue}>
      {isDrawerOpen !== null && (
        <DrawerPanel
          isOpen={isDrawerOpen}
          onClose={handleClose}
          onOpen={options?.onOpen}
          closeOnOutsideClick={closeOnOutsideClick}
          closeOnEscapeKeypress={closeOnEscapeKeypress}
        >
          {renderedChild}
        </DrawerPanel>
      )}
      {children}
    </DrawerContext.Provider>
  );
```

Probably a bit cleaner, but now it's even more questionable why `isOpen` exists lol

Reply:
It lingered from the components I mimicked from the starfish implementation. Those are always rendered so they control whether or not they're expanded. I've removed it for now but still have to do the animation fix when it unmounts :+1: